services/application.go

```diff
@@ -133,3 +140,65 @@ func (app *ChainlinkApplication) RemoveAdapter(bt *models.BridgeType) error {
 
 	return nil
 }
+
+// RunManager safely handles coordinating job runs.
+type RunManager struct {
+	store  *store.Store
+	waiter sync.WaitGroup
+}
+
+// NewRunManager initializes a RunManager.
+func NewRunManager(store *store.Store) *RunManager {
+	return &RunManager{
+		store: store,
+	}
+}
+
+func (rm *RunManager) Start() error {
+	if err := rm.ResumeSleepingRuns(); err != nil {
+		return err
+	}
+	rm.executeRunQueue()
+
+	return nil
+}
+
+func (rm *RunManager) ResumeSleepingRuns() error {
+	pendingRuns, err := rm.store.JobRunsWithStatus(models.RunStatusPendingSleep)
+	if err != nil {
+		return err
+	}
+	for _, run := range pendingRuns {
+		go func(run models.JobRun) {
```

Comment:
Could this just use the `RunQueue.Push` to be executed in the method below? `executeRunQueue`